### PR TITLE
bluespace vendor fix

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
fix issue with bluespace vendor where if the gas you buying runs out while filling a tank, the machine would runtime and be stuck forever.

## Why It's Good For The Game
issue fix

## Changelog
:cl:
fix: bluespace vendor no longer get stuck when the gas you filling with finishes midway
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/machinery/bluespace_vendor.dm [hunk 1]
```diff
@@ -112,6 +112,20 @@
 	if(!selected_gas)
 		return
 	var/gas_path = gas_id2path(selected_gas)
+
+	var/list/gases = connected_machine.bluespace_network.gases
+	var/gas_found = FALSE
+	for(var/gas_id in gases)
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/bluespace_vendor.dm:118]: Why are you doing a for loop here, you already have the path you want to check don't you?

Comment (Ghilker) [reply code/modules/atmospherics/machinery/bluespace_vendor.dm:118]: So I just do `if(!connected_machine.bluespace_network.gases[gas_path])`?

Comment (LemonInTheDark) [reply code/modules/atmospherics/machinery/bluespace_vendor.dm:118]: I don't see why not

Comment (Ghilker) [reply code/modules/atmospherics/machinery/bluespace_vendor.dm]: done

**Pull request merged**